base/bucket_gocb.go

```diff
@@ -147,6 +152,72 @@ func (bucket CouchbaseBucketGoCB) GetBucketCredentials() (username, password str
 	return bucket.spec.BucketName, password
 }
 
+func (bucket CouchbaseBucketGoCB) GetMetadataPurgeInterval() (int, error) {
+
+	var err error
+
+	// Check for Bucket-specific setting first
+	bucketReqUri := fmt.Sprintf("%s/pools/default/buckets/%s", bucket.spec.Server, bucket.spec.BucketName)
+
+	bucketPurgeInterval, err := bucket.RetrievePurgeInterval(bucketReqUri)
+	if bucketPurgeInterval > 0 || err != nil {
+		return bucketPurgeInterval, err
+	}
+
+	// Cluster-wide settings
+	clusterReqUri := fmt.Sprintf("%s/settings/autoCompaction", bucket.spec.Server)
+	clusterPurgeInterval, err := bucket.RetrievePurgeInterval(clusterReqUri)
+	if clusterPurgeInterval > 0 || err != nil {
+		return clusterPurgeInterval, err
+	}
+
+	return 0, nil
+
+}
+
+// Retrieves Metadata Purge Interval from server and converts to hours.  Uses bucket purge interval
+// if specified - otherwise uses cluster purge interval.
+func (bucket CouchbaseBucketGoCB) RetrievePurgeInterval(uri string) (int, error) {
+
+	// Both of the purge interval endpoints (cluster and bucket) return purgeInterval in the same way
+	var purgeResponse struct {
+		PurgeInterval float64 `json:"purgeInterval,omitempty"`
+	}
+
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		return 0, err
+	}
+	username, password, _ := bucket.spec.Auth.GetCredentials()
+	req.SetBasicAuth(username, password)
+
+	client := bucket.Bucket.IoRouter()
+	resp, err := client.HttpClient().Do(req)
+	if err != nil {
+		return 0, err
+	}
+
+	jsonDec := json.NewDecoder(resp.Body)
+	defer resp.Body.Close()
+	err = jsonDec.Decode(&purgeResponse)
+	if err != nil {
+		return 0, err
+	}
+
+	if resp.StatusCode == http.StatusForbidden {
+		Warn("403 Forbidden attempting to access %s.  Bucket user must have Bucket Full Access and Bucket Admin roles to retrieve metadata purge interval.", uri)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return 0, errors.New(resp.Status)
+	}
+
+	// Server purge interval is a float value, in days.  Round up to hours
+	purgeIntervalHours := int(purgeResponse.PurgeInterval*24 + 0.5)
```

Comment:
Can you pull out `0.5` into a named variable? 

```
// Give some padding because ...
paddingHours := 0.5
purgeIntervalHours := int(purgeResponse.PurgeInterval*24 + paddingHours)
```